flake.nix

```diff
@@ -0,0 +1,62 @@
+{
+  description = "WIP stuff to get started on the pinebook pro.";
+
+  inputs = {
+    nixpkgs.url = "github:NixOS/nixpkgs-channels/nixos-unstable";
+    flake-utils.url = "github:numtide/flake-utils";
+  };
+
+  outputs = { self, nixpkgs, flake-utils }:
+    {
+      overlay = final: prev: import "${self}/overlay.nix" final prev;
+    } // flake-utils.lib.eachDefaultSystem (
+      system:
+        let
+          inherit (nixpkgs) lib;
+          darwin_network_cmds_openssl_overlay = final: prev: {
+            darwin = prev.darwin // {
+              network_cmds = prev.darwin.network_cmds.override { openssl_1_0_2 = prev.openssl; };
+            };
+          };
+          pkgsNative = import nixpkgs {
+            inherit system;
+            config.allowUnfree = true;
+            overlays = [ darwin_network_cmds_openssl_overlay self.overlay ];
+          };
+          pkgs = if system == "aarch64-linux" then pkgsNative else pkgsNative.pkgsCross.aarch64-multiplatform;
+        in
+          rec {
+            packages = rec {
+              inherit (pkgs)
+                uBootPinebookPro uBootPinebookProExternalFirst
+                linux_pinebookpro_latest
+                linux_pinebookpro_lts
```

Comment:
I don't know what's expected to be provided through the `packages` attribute of a flake, but I think it's possible we want to also provide the matching `linuxPackages_*` attrsets.

Reply:
Right, at first I included them, but `nix flakes check` wanted all of the things in `packages` to be derivations, not sets of derivations. I can't remember if it was an error or a warning though, so we might be able to ignore it, and if not we can include it as a separate output outside of `packages`. Basically `nix flakes` understands certain attributes of `outputs` and will check their types, but we can include whatever we want if it's being annoying about `packages`.